test/jasmine/assets/custom_assertions.js

```diff
@@ -37,16 +37,28 @@ exports.assertStyle = function(dims, color, opacity) {
             .toEqual(dims[i], 'to have correct number of pts in trace ' + i);
 
         points.each(function() {
-            var point = d3.select(this);
-
-            expect(point.style('fill'))
+            expect(this.style.fill)
                 .toEqual(color[i], 'to have correct pt color');
-            expect(+point.style('opacity'))
+            var op = this.style.opacity;
+            expect(op === undefined ? 1 : +op)
                 .toEqual(opacity[i], 'to have correct pt opacity');
         });
     });
 };
 
+exports.assertHoverLabelStyle = function(g, expectation, msg, textSelector) {
+    if(!msg) msg = '';
+
+    var path = g.select('path').node();
+    expect(getComputedStyle(path).fill).toBe(expectation.bgcolor, msg + ': bgcolor');
```

Comment:
`window.getComputedStyle`

Reply:
... but this is in a test file. I don't care about that as much. Your call.

Reply:
updated them all -> https://github.com/plotly/plotly.js/pull/2046/commits/78f5222c9c93e875eb758738d13ccd1054c0e549

tested that we only call `getComputedStyle` on `window` and only as many times as expected -> https://github.com/plotly/plotly.js/pull/2046/commits/97cea609b9fb75f5cec9e40893497b4b831cc0d2